### PR TITLE
[WIP] Analytics for each page

### PR DESCRIPTION
**Pull request by choiboy98**
All pages work except for the Apply page.
Apply page does not seem to work because it does not have a class on its own. I could not figure out how to add Google Analytics code to the const function.

The new google analytics is done with uiuc@hack4impact.org email.

### CODE REVIEW

File: components/Analytics.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import ReactGA from 'react-ga';
+
+const initializeGA = () => {
+    ReactGA.initialize('UA-144770649-1');
+}
+
+const logPage = () => {
+    //ReactGA.pageview(window.location.pathname + window.location.search);
```
Comment (angadgarg25) [inline components/Analytics.js]: lets remove this?

Comment (angadgarg25) [inline components/Analytics.js]: lets remove this comment

File: pages/_app.js [hunk 2]
```diff
@@ -5,11 +5,14 @@ import Nav from "../components/nav";
...
 import Footer from "../components/footer";
 import { PageTransition } from "next-page-transitions";
+import { initializeGA, logPage, logEvent } from "../components/Analytics"
 
 export default class MyApp extends App {
   constructor(props) {
     super(props);
     this.state = { hasError: false };
+    // initializeGA()
+    // logPage()
```
Comment (angadgarg25) [inline pages/_app.js]: remove comments

Comment (angadgarg25) [reply pages/_app.js]: and remove the imports for these if they're not used anywhere else in this file

File: pages/index.js [hunk 3]
```diff
@@ -19,8 +19,9 @@ class Home extends React.Component {
     //   className: "announcement",
     //   bodyClassName: "announcement-body"
     // });
-    ReactGA.initialize('UA-143288182-1');
-    ReactGA.pageview('/index');
+    initializeGA()
+    logPage()
+
```
Comment (angadgarg25) [inline pages/index.js]: remove empty line

**Review by angadgarg25 — CHANGES_REQUESTED** [review]
Mostly nits. I'll take a look at fixing apply page today. Also could you remove  `logEvent` from your imports for now? As far as I can tell, we don't currently use it yet, once we want to use it we can add it back in.

File: pages/apply/index.js [hunk 4]
```diff
@@ -5,10 +5,6 @@ import Nav from "../../components/nav";
 import { Container, Row, Col } from "reactstrap";
 import Link from "next/link";
 import Footer from "../../components/footer";
-import ReactGA from 'react-ga';
-
-ReactGA.initialize('UA-143288182-1');
-ReactGA.pageview('/apply');
 
 const ApplyPage = () => (
```
Comment (angadgarg25) [inline pages/apply/index.js]: this is a short hand way to define react components, so you can't define `componentDidMount`. Assuming thats why you couldn't add google analytics, all you have to do is convert this to look like a regular component.
`class ApplyPage extends React.Component {` and then move the jsx code to be returned by a render function.

**Review by angadgarg25 — DISMISSED** [review]
🐓

**Review by rgychiu — APPROVED** [review]
Looks good to me based on Angad's previous comments as well

**Pull request merged**